packages/ra-ui-materialui/src/button/SaveButton.tsx

```diff
@@ -88,13 +93,19 @@ export const SaveButton = <RecordType extends RaRecord = any>(
     );
 
     const handleSubmit = useCallback(
-        async values => {
+        async (values, event) => {
             let errors;
-            if (saveContext?.save) {
-                errors = await saveContext.save(values, {
+            const finalValues = sanitizeEmptyValues
+                ? sanitizeValues(values, record)
+                : values;
+
+            if ((mutationOptions || transform) && saveContext?.save) {
```

Comment:
If mutationOptions || transform are getting passed, the SaveContext will be used first

Reply:
I think there is 1 case not covered by this new logic: if you provide neither `mutationOptions` nor `transform` nor `onSubmit`, then you should still call the `SaveContext` if there is one. There should be a 3rd `else if` IMO.